tests/unit/test_args.py

```diff
@@ -529,6 +529,90 @@ def test_doesnt_create_ruleset_resolver_if_not_given_data(self):
+        self.config_store.set_config_variable(
+            'request_min_compression_size_bytes', 100
+        )
+        self.config_store.set_config_variable(
+            'disable_request_compression', True
+        )
+        config = self.call_get_client_args()['client_config']
+        self.assertEqual(config.request_min_compression_size_bytes, 100)
+        self.assertTrue(config.disable_request_compression)
+
+    def test_request_compression_client_config_overrides_config_store(self):
+        self.config_store.set_config_variable(
+            'request_min_compression_size_bytes', 100
+        )
+        self.config_store.set_config_variable(
+            'disable_request_compression', True
+        )
+        config = self.call_get_client_args(
+            client_config=Config(
+                disable_request_compression=False,
+                request_min_compression_size_bytes=0,
+            )
+        )['client_config']
+        self.assertEqual(config.request_min_compression_size_bytes, 0)
+        self.assertFalse(config.disable_request_compression)
+
+    def test_bad_type_request_min_compression_size_bytes(self):
+        with self.assertRaises(exceptions.InvalidConfigError):
+            self.call_get_client_args(
+                client_config=Config(
+                    request_min_compression_size_bytes='foo',
+                ),
+            )
+        self.config_store.set_config_variable(
+            'request_min_compression_size_bytes', 'foo'
+        )
+        with self.assertRaises(exceptions.InvalidConfigError):
+            self.call_get_client_args()
+
+    def test_low_min_request_min_compression_size_bytes(self):
+        with self.assertRaises(exceptions.InvalidConfigError):
+            self.call_get_client_args(
+                client_config=Config(
+                    request_min_compression_size_bytes=-1,
+                ),
+            )
+        self.config_store.set_config_variable(
+            'request_min_compression_size_bytes', -1
+        )
+        with self.assertRaises(exceptions.InvalidConfigError):
+            self.call_get_client_args()
+
+    def test_high_max_request_min_compression_size_bytes(self):
+        with self.assertRaises(exceptions.InvalidConfigError):
+            self.call_get_client_args(
+                client_config=Config(
+                    request_min_compression_size_bytes=9999999,
+                ),
+            )
+        self.config_store.set_config_variable(
+            'request_min_compression_size_bytes', 9999999
+        )
+        with self.assertRaises(exceptions.InvalidConfigError):
+            self.call_get_client_args()
+
+    def test_bad_value_disable_request_compression(self):
+        config = self.call_get_client_args(
+            client_config=Config(
+                disable_request_compression='foo',
+            ),
+        )['client_config']
+        self.assertFalse(config.disable_request_compression)
```

Comment:
This test suite also uses an empty `ConfigValueStore` object. Because of that, this test would not use the `ensure_boolean` conversion function defined in the config provider. Maybe as a followup we should look into running these tests as well as the ones in `test_client.TestAutoGeneratedClient` using a config value store that contains the default config mapping.